### PR TITLE
Split date skeleton key in two, with datagen

### PR DESCRIPTION
**Pull request by sffc**
Need to see if this helps with data size

### CODE REVIEW

**Comment by sffc** [comment]
Sizes:

```
164K    /tmp/icu4x_both.postcard
96K     /tmp/icu4x_day.postcard
52K     /tmp/icu4x_ext.postcard
```

So it looks like splitting does make things smaller.

File: components/datetime/src/calendar.rs [hunk 1]
```diff
@@ -70,8 +70,12 @@ pub trait CldrCalendar: InternalCldrCalendar {
     type DatePatternV1Marker: KeyedDataMarker<Yokeable = DatePatternV1<'static>>;
 
     #[cfg(any(feature = "datagen", feature = "experimental"))]
-    /// The data marker for loading date pattern skeletons for this calendar.
-    type DateSkeletonPatternsV1Marker: KeyedDataMarker<Yokeable = PackedSkeletonDataV1<'static>>;
+    /// The data marker for loading date-day pattern skeletons for this calendar.
+    type DateDaySkeletonPatternsV1Marker: KeyedDataMarker<Yokeable = PackedSkeletonDataV1<'static>>;
+
+    #[cfg(any(feature = "datagen", feature = "experimental"))]
+    /// The data marker for loading date-ext pattern skeletons for this calendar.
```
Comment (Manishearth) [inline components/datetime/src/calendar.rs:77]: we should have more comments explaining what ext is

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
I did a little [experiment](https://docs.google.com/spreadsheets/d/1YHXcpGgT9V1pF-xaa1nDZNwX9GkXrOPgxXVG5iE5ioI/edit) to find the most efficient split.

For splitting into 2 keys, we're near the optimum with the split proposed in this PR.

For splitting into 3 keys, we should add a split between YearMonthDay and EraYearMonthDay.

For splitting into 4 keys, we should add a split between Year and EraYear.

It's also possible that reordering the skeleta within the list would yield smaller blocks. I haven't tested this.

**Comment by sffc** [comment]
Splitting the skeleta each into a separate key makes things the smallest, even with the lookup tables. The lookup tables get smaller when the keys are smaller because there is more deduplication. Even in hybrid mode, the wins on data size are great enough that it looks like it would more than offset the increase in lookup table sizes.

**Comment by sffc** [comment]
Here's a hypothesis worth testing. I expect that making the keys shared across calendars but unique on components would yield better results than the other way around. Patterns don't differ as much across calendars as display names, but they differ a lot from skeleton to skeleton.

**Comment by sffc** [comment]
Moving this to draft because I want to compare it with #4779

**Comment by Manishearth** [comment]
>  I expect that making the keys shared across calendars but unique on components would yield better results than the other way around.

Wait, really? How would we handle cases where calendars differ, an "extra values" table?

**Comment by sffc** [comment]
Code archived at https://github.com/sffc/omnicu/tree/archive/neo-skeleta-two-key

**Pull request closed without merging**